lib/parsec/upgrade.py

```diff
@@ -176,7 +176,21 @@ def upgrade(self):
                             warnings[vn].append(msg)
                         self.del_item(upg['old'])
                         if upg['cvt'].describe() != "DELETED (OBSOLETE)":
-                            self.put_item(upg['new'], upg['cvt'].convert(old))
+                            # check self.cfg does not already contain a
+                            # non-deprecated item matching upg['new']:
+                            try:
+                                self.get_item(upg['new'])
+                            except KeyError:
+                                self.put_item(upg['new'],
+                                              upg['cvt'].convert(old))
+                            else:
+                                errMsg = (
+                                    'ERROR: Cannot perform deprecation '
+                                    'upgrade "' + msg + '" because '
+                                    'deprecated item clashes with existing '
+                                    'non-deprecated item'
```

Comment:
I think this text could be clearer. 
```suggestion
                                errMsg = (
                                    'ERROR: Cannot upgrade deprecated '
                                    'item "' + msg + '" because the upgraded '
                                    'item already exists'
```

Reply:
Yes, I was struggling to word it well